File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/states/teleop/examples/DriveTeleop.java
package org.firstinspires.ftc.teamcode.teamcode.states.teleop.examples;


import org.firstinspires.ftc.teamcode.teamcode.hardware.RobotHardware;
import org.firstinspires.ftc.teamcode.teamcode.stateengine.State;

public class DriveTeleop implements State {

    RobotHardware rh = null;

    @Override
    public void init(RobotHardware rh) {
        this.rh = rh;
    }

    @Override
    public void run() {

        double max     =  rh.controls.exampleDriveSpeed();

        // POV Mode uses left joystick to go forward & strafe, and right joystick to rotate.
        double axial   =  rh.controls.driveY();
        double lateral =  rh.controls.driveX();
        double yaw     =  rh.controls.driveR();

        rh.exampleDriveH.drive(axial, lateral, yaw, max);
    }

    @Override
    public void stop() {

    }

    @Override
    public boolean isDone() {
        return false;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/states/teleop/examples/DriveTeleop.kt
package org.firstinspires.ftc.teamcode.states.teleop.examples

import org.firstinspires.ftc.teamcode.hardware.RobotHardware
import org.firstinspires.ftc.teamcode.stateengine.State

class DriveTeleop() : State {
  private lateinit var rh: RobotHardware

  override fun init(rh: RobotHardware) {
    this.rh = rh
  }

  override fun run() {
    val max = rh.controls.exampleDriveSpeed()

    val axial = rh.controls.driveY()
    val lateral = rh.controls.driveX()
    val yaw = rh.controls.driveR()

    rh.driveH.drive(axial, lateral, yaw, max)
  }

  override val isDone = false
}
